pkg/reconciler/ingress/resources/secret_ce.go

```diff
@@ -33,7 +34,24 @@ func CalculateCertificateHash(secret *corev1.Secret) (string, error) {
 
 // MakeMirrorSecret creates a Secret object that mirrors a TLS Secret
 func MakeMirrorSecret(originSecret *corev1.Secret, certificateHash string) *corev1.Secret {
+	removeEmptyLinesSecret(originSecret)
 	return makeSecret(originSecret, certificateHash, config.IstioNamespace, map[string]string{
 		labelKeyDomainMappingSecret: labelValueDomainMappingSecret,
 	}, map[string]string{})
 }
+
+func removeEmptyLines(data []byte) []byte {
+	var cleanedLines []string
+	lines := strings.Split(string(data), "\n")
+	for _, line := range lines {
+		if strings.TrimSpace(line) != "" {
+			cleanedLines = append(cleanedLines, line)
+		}
+	}
+	return []byte(strings.Join(cleanedLines, "\n"))
```

Comment:
It is just a gut feeling, but this feels off. I wouldn't touch the content just by using string functions. Have a look whether https://pkg.go.dev/encoding/pem with its `Decode` and `Encode` do work here.

Reply:
Got it!! used pem decode and encode as you suggested